users/service.go

```diff
@@ -180,24 +180,38 @@ func (svc service) ListClients(ctx context.Context, token string, pm mgclients.P
 	if err != nil {
 		return mgclients.ClientsPage{}, err
 	}
-	if err := svc.checkSuperAdmin(ctx, userID); err == nil {
-		pm.Role = mgclients.AllRole
-		pg, err := svc.clients.RetrieveAll(ctx, pm)
-		if err != nil {
-			return mgclients.ClientsPage{}, errors.Wrap(svcerr.ErrViewEntity, err)
-		}
-		return pg, err
+	if err := svc.checkSuperAdmin(ctx, userID); err != nil {
+		return mgclients.ClientsPage{}, errors.Wrap(svcerr.ErrViewEntity, err)
 	}
```

Comment:
`svc.checkSuperAdmin` have already wrapped with service error 
No need for wrap with error again here.
```suggestion
		return mgclients.ClientsPage{}, err
```